### ISSUE TITLE
FileTabCharacter: update  doc to mention  that only first violation is reported for default configuration

### ISSUE THREAD

**Issue by ElinaZoldnere** [checkstyle/checkstyle#16225]
I encountered this issue while working on enabling example tests for SuppressWithPlainTextCommentFilter.

The issue: `FileTabCharacter` check with default configuration (reporting only the first violation) does not report a file if that first violation is suppressed, even when there are more actual violations, and the user probably would expect checking the rest of the file.
https://checkstyle.sourceforge.io/checks/whitespace/filetabcharacter.html

```
$ cat ./Test.sql
-- stop check
-- suppressed violation below
	SELECT * FROM users
-- resume check
-- expected violation below
	SELECT 1
```

```
$ cat config-with-suppression.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="FileTabCharacter"/>

    <module name="SuppressWithPlainTextCommentFilter">
        <property name="offCommentFormat" value="stop check"/>
        <property name="onCommentFormat" value="resume check"/>
    </module>

</module>
```

```
$ java -jar checkstyle-10.21.1-all.jar -c config-with-suppression.xml Test.sql
Starting audit...
Audit done.
```

If `FileTabCharacter` check  is configured to report each line, it reports a violation on line 6.
```
$ cat config-with-suppression-eachline.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="FileTabCharacter">
        <property name="eachLine" value="true"/>
    </module>

    <module name="SuppressWithPlainTextCommentFilter">
        <property name="offCommentFormat" value="stop check"/>
        <property name="onCommentFormat" value="resume check"/>
    </module>

</module>
```

```
$ java -jar checkstyle-10.21.1-all.jar -c config-with-suppression-eachline.xml Test.sql
Starting audit...
[ERROR] /home/elina/IdeaProjects/sandbox/jdk17/FileTabCharacter-issue/Test.sql:6:1: Line contains a tab character. [FileTabCharacter]
Audit done.
Checkstyle ends with 1 errors.
```

I am not sure whether this qualifies as a bug to fix or a limitation to tolerate, but this behaviour should at least be documented.

**Comment by romani** [comment]
Thanks a lot for reporting this.

I don't know initial goals, but I think it is done on purpose. Imagine you run this Check on sources that are tab formatted, there will be uncountable amount of violations on small projects, it will blow up report.

But what Check is doing is pointing on file where tab is used, so user can adjust his formatter to do spaces. I doubt that we'll have real live use case where first tab is ok and all others should be reported :).

But I agree documentation should be extended to state it explicitly.

**Comment by romani** [comment]
Please send PR !

**Comment by rnveach** [comment]
https://checkstyle.org/checks/whitespace/filetabcharacter.html#Properties
> Control whether to report on each line containing a tab, or just the first instance.

Seems like this is already documented.

**Comment by ElinaZoldnere** [comment]
I feel I didn't manage to explain my point.
The configuration itself is clear. But something that I would probably like to see there regarding the default configuration, is an explicit note to be aware of - something like: *"if the first instance falls under a suppression rule, the file will not be reported"*.

I can think of a situation where someone uses a general suppression (although probably not recommended) for section of the code for some other obvious and trivial reason, and can miss the entire tab-formatted file if the first instance happens to fall within the suppressed section. Moreover, this violation is not visible.

But probably it is too theoretical situation, or this is too obvious for a typical user, that adding anything more is redundant.

**Comment by romani** [comment]
Please send PR , let's see how it will suit in documentation, if good, we can merge.

**Comment by ElinaZoldnere** [comment]
@romani 
Thanks, but only if you both approve. :)

I explained how it looks from my point of view, but you decide whether it’s relevant in the overall context.

**Comment by romani** [comment]
Please send PR, it is better to see exact wording on proper placement to see a reason

**Comment by rnveach** [comment]
> but only if you both approve

I will approve because of this comment. This is more than I get now then from someone else.

**Comment by romani** [comment]
Clarification is provided without reference to filter.

It is unusual , in comparison  to most others Checks, behavior that by default only first is reported. So extra wording is good.
